docs/java/how-to-set-child-workflow-options-in-java.md

```diff
@@ -0,0 +1,252 @@
+                WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE
+        )
+        .build();
+        // Get the Child Workflow stub
+        ChildWorkflow child = Workflow.newChildWorkflowStub(ChildWorkflow.class, childWorkflowOptions);
+        // invoke Child Workflow and wait for it to complete
+        child.executeChild();
+    }
+```
+
+See [What is a Workflow Id Reuse Policy?](/docs/concepts/what-is-a-workflow-id-reuse-policy)
+
+### `WorkflowExecutionTimeout`
+
+- Type: `time.Duration`
+- Default: Unlimited
+
+```java
+ private void parentWorkflow() {
+        ChildWorkflowOptions childWorkflowOptions =
+                ChildWorkflowOptions.newBuilder()
+                        .setWorkflowExecutionTimeout(Duration.ofSeconds(10))
+                        .build();
+        // Get the Child Workflow stub
+        ChildWorkflow child = Workflow.newChildWorkflowStub(ChildWorkflow.class, childWorkflowOptions);
+        // invoke Child Workflow and wait for it to complete
+        child.executeChild();
+    }
+```
+
+See [What is a Workflow Execution Timeout?](/docs/concepts/what-is-a-workflow-execution-timeout)
+
+### `WorkflowRunTimeout`
+
+- Type: `time.Duration`
+- Default: Same as [WorkflowExecutionTimeout](#WorkflowExecutionTimeout).
+
+```java
+private void parentWorkflow() {
+        ChildWorkflowOptions childWorkflowOptions =
+                ChildWorkflowOptions.newBuilder()
+                        .setWorkflowRunTimeout(Duration.ofSeconds(4))
+                        .build();
+        // Get the Child Workflow stub
+        ChildWorkflow child = Workflow.newChildWorkflowStub(ChildWorkflow.class, childWorkflowOptions);
+        // invoke Child Workflow and wait for it to complete
+        child.executeChild();
+    }
+```
+
+See [What is a Workflow Run Timeout?](/docs/concepts/what-is-a-workflow-run-timeout)
+
+### `WorkflowTaskTimeout`
+
+- Type: `time.Duration`
+- Default: 10 seconds.
+- Values: Maximum accepted value is 60 seconds.
+
+```java
+ private void parentWorkflow() {
+        ChildWorkflowOptions childWorkflowOptions =
+                ChildWorkflowOptions.newBuilder()
+                        .setWorkflowTaskTimeout(Duration.ofSeconds(10))
+                        .build();
+        // Get the Child Workflow stub
+        ChildWorkflow child = Workflow.newChildWorkflowStub(ChildWorkflow.class, childWorkflowOptions);
+        // invoke Child Workflow and wait for it to complete
+        child.executeChild();
+    }
+```
+
+See [What is a Workflow Task Timeout?](/docs/concepts/what-is-a-workflow-task-timeout)
+
+### `RetryOptions`
+
+- Type: `RetryOptions`
+- Default: `Null` which means no retries will be attempted.
+
+```java
+private static void parentWorkflow() {
+        ChildWorkflowOptions childworkflowOptions =
+                ChildWorkflowOptions.newBuilder()
+                        .setRetryOptions(RetryOptions.newBuilder()
+                                .setMaximumAttempts(2)
```

Comment:
would recommend not showing users maximumAttempts, but rather set a execution timeout in child workflowoptions

Reply:
updated!